src/Vlingo.Actors/Plugin/Mailbox/SharedRingBuffer/RingBufferDispatcher.cs

```diff
@@ -38,12 +40,15 @@ public void Execute(IMailbox mailbox)
 
         public void Start()
         {
-            if (started != null)
+            lock (mutex)
             {
-                return;
+                if (started != null)
+                {
+                    return;
+                }
+
+                started = Task.Run(() => Run(), cancellationTokenSource.Token);
```

Comment:
I should have asked this earlier. But, the cancellationToken you are using, it is actually being canceled in `Execute(...)` method. As far as I remember, this is to replace the `thread.Interrupt()` method from the previous code. I am not quite sure if canceling cancellationToken is providing the same behavior. Would you consider giving it some thoughts?